### ISSUE TITLE
False detection of AS0022 when function is obsolete

### ISSUE THREAD

**Issue by jwikman** [microsoft/AL#5870]
**Describe the bug**
This is a variant of #5869 
When a procedure in a page, report or table has the Obsolete attribute and then is changed to internal AS0022 occurs.

**To Reproduce**
Steps and to reproduce the behavior:
1. Create a page with a couple of procedures
2. Set ObsoleteState attribute on the procedures
3. Compile to get an app file
4. Copy the app file to the .alpackages folder
5. Configure appsourcecop.json for that version
6. Increase version in app.json
7. Change one procedure to internal and compile

Now AS0022 will appear.

v1.0.0.0
```
page 50100 "asdf Test"
{
    PageType = Card;
    SourceTable = Customer;
    Caption = 'asdf Test';
    UsageCategory = None;

    layout
    {
        area(content)
        {
            group(General)
            {
                field("No."; "No.")
                {
                    ApplicationArea = All;
                }
                field(Name; Name)
                {
                    ApplicationArea = All;
                }
            }
        }
    }

    [Obsolete('will be removed', '15.0')]
    procedure MyProcedure()
    begin
        Message('repro');
    end;

    [Obsolete('will be removed', '15.0')]
    procedure MyProcedure2()
    begin
        Message('now');
    end;

}
```
v1.1.0.0
```
page 50100 "asdf Test"
{
    PageType = Card;
    SourceTable = Customer;
    Caption = 'asdf Test';
    UsageCategory = None;

    layout
    {
        area(content)
        {
            group(General)
            {
                field("No."; "No.")
                {
                    ApplicationArea = All;
                }
                field(Name; Name)
                {
                    ApplicationArea = All;
                }
            }
        }
    }

    internal procedure MyProcedure()
    begin
        Message('repro');
    end;

    [Obsolete('will be removed', '15.0')]
    procedure MyProcedure2()
    begin
        Message('now');
    end;

}
```

**Expected behavior**
We should be able to change the scope of a procedure if the procedure (or object) was obsolete pending.

**Screenshots**
N/A

**5. Versions:**

 - AL Language: 5.0.270354

 - Business Central: N/A

**Comment by qutreson** [comment]
Hi @jwikman, this seems like a scenario that should be allowed by the AppSourceCop. When a procedure (or an object in general) is obsolete in the base line version, but is not obsolete anymore in the current extension, the ApppSourceCop should consider the procedure (or the object) as a brand new one which is not part of the public interface. We should then not report diagnostics for breaking changes.

We have fixed similar issues on some AppSourceCop rules, but maybe we did not fix this one.

**Comment by qutreson** [comment]
@jwikman we just published a new version of the AL extension 5.0.271898, it should fix issues related to this scenario, can you give it a try?

**Comment by jwikman** [comment]
@qutreson I will try to repro on your brand new version :)

**Comment by jwikman** [comment]
@qutreson I can confirm that this one is fixed in 5.0.271898
Thanks!
